package.json

```diff
@@ -33,6 +33,7 @@
     "bootstrap": "./scripts/bootstrap.sh  && ./scripts/link-dependencies.sh",
     "build": "yarn nx run-many -t=build",
     "build:dev": "lerna run --stream prebuild && yarn nx run-many --target=build --output-style=dynamic --watch --preserveWatchOutput",
+    "check:types": "lerna run check:types",
```

Comment:
The normal build already checks its types, so I did not want to duplicate steps. Maybe it would be cleaner to add the `check:types` to all the packages, but then it would be incoherent with the "non-typed" packages

Reply:
However that might not be possible, the type checking for a module that relies on another module would always require that `.d.ts` files have been produced

Reply:
One option might be to use a top level file `tsconfig.json`

```json
{
  "references": [
    { "path": "./packages/types" },
    { "path": "./packages/backend-core" },
    { "path": "./packages/server" },
    { "path": "./packages/worker" }
   ...etc
  ],
  "files" :[]
}
```

As the target for `tsc -p`. This would run the typescript build in order for all the dependencies listed. Provided that the `references` of each sub `package.json` includes the packages that are depended on. 

This file is currently in master, although it looks to have been removed in `develop`. This is actually a required file for `yarn build:dev`